src/protocol/logs.rs

```diff
@@ -0,0 +1,212 @@
+pub mod events {
+    use serde::{Deserialize, Serialize};
+    use crate::protocol::runtime::methods::{StackTrace, RemoteObject};
+    use crate::protocol::types::{JsInt};
+
+    #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+    #[serde(rename_all = "lowercase")]
+    pub enum  LogEntrySource {
+        XML,
+        JAVASCRIPT,
+        NETWORK,
+        STORAGE,
+        APPCACHE,
+        RENDERING,
+        SECURITY,
+        DEPRECATION,
+        WORKER,
+        VIOLATION,
+        INTERVENTION,
+        RECOMMENDATION,
+        OTHER,
+    }
+
+    #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+    #[serde(rename_all = "lowercase")]
+    pub enum  LogEntryLevel {
+        VERBOSE,
+        INFO,
+        WARNING,
+        ERROR,
+    }
+
+    /// Log entry for a logs
+    /// See https://chromedevtools.github.io/devtools-protocol/tot/Log#type-LogEntry
+    #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+    #[serde(rename_all = "camelCase")]
+    pub struct LogEntry {
+        pub source: LogEntrySource,
+        pub level: LogEntryLevel,
+        pub text: String,
+        /// Timestamp when this entry was added
+        /// Number of milliseconds since epoch
+        /// See https://chromedevtools.github.io/devtools-protocol/tot/Runtime#type-Timestamp
+        pub timestamp: f64,
```

Comment:
Maybe we want something from `chrono` here so there is no ambiguity.

Reply:
Basically I used the same type for timestamp as in `ResponseReceivedEventParams` from `network.rs`, but in chrome it means a little bit different, but represents still with the same `number` type.

Reply:
I'm open to using chrono for these fields later on, but at the moment the other timestamps are all floats, so I don't think this should block the PR